api_tests/users/views/user_profile_test_mixin.py

```diff
@@ -0,0 +1,374 @@
+        set_user_profile_order(user_profile_object_ids[::-1])
+        res = app.get(list_url, auth=user.auth)
+        profile_institutions = [each['attributes']['institution'] for each in res.json['data']]
+        assert len(profile_institutions) == 3
+        assert '3' in profile_institutions[0]
+        assert '2' in profile_institutions[1]
+        assert '1' in profile_institutions[2]
+
+
+@pytest.mark.django_db
+class UserProfileDetailMixin(UserProfileFixtures):
+
+    @pytest.fixture
+    def detail_url(self):
+        raise NotImplementedError
+
+    def test_get_detail(self, app, user, user_two, detail_url):
+        # unauthoized can access
+        res = app.get(detail_url)
+        assert res.status_code == 200
+
+        # another authorized user can access
+        other_user = AuthUserFactory()
+        res = app.get(detail_url, auth=other_user.auth)
+        assert res.status_code == 200
+
+        # authorized can access self
+        res = app.get(detail_url, auth=user.auth)
+        assert res.status_code == 200
+
+
+@pytest.mark.django_db
+class UserProfileCreateMixin(UserProfileFixtures):
+
+    def test_create_profile_object_fails(self, app, list_url, user, user_two, payload):
+        # no auth fails
+        res = app.post_json(list_url, payload(institution='Woop!'), expect_errors=True)
+        assert res.status_code == 401
+
+        # auth but not the user fails
+        res = app.post_json(list_url, payload(institution='Helloooo'), auth=user_two.auth, expect_errors=True)
+        assert res.status_code == 403
+
+        # wrong type fails
+        wrong_type = payload(institution='Woop!')
+        wrong_type['data']['type'] = 'snails'
+        res = app.post_json(list_url, wrong_type, auth=user.auth, expect_errors=True)
+        assert res.status_code == 409
+
+        # Institution is empty fails
+        inst_empty = payload(institution=None)
+        res = app.post_json(list_url, inst_empty, auth=user.auth, expect_errors=True)
+        assert res.status_code == 400
+
+        # start_date no end_date but not ongoing fails
+        wrong_dates = payload(institution='Well lets go', start_date='2018-01-01', end_date=None, ongoing=False)
+        res = app.post_json(list_url, wrong_dates, auth=user.auth, expect_errors=True)
+        assert res.status_code == 400
+
+        # end_date but no start_date fails
+        wrong_dates = payload(institution='Yeah!', end_date='2018-01-01', start_date=None)
+        res = app.post_json(list_url, wrong_dates, auth=user.auth, expect_errors=True)
+        assert res.status_code == 400
+
+        # end_date before start_date fails
+        flopped_dates = payload(institution='What', start_date='2019-01-01', end_date='2018-01-01')
+        res = app.post_json(list_url, flopped_dates, auth=user.auth, expect_errors=True)
+        assert res.status_code == 400
+
+        # ongoing with start and end date fails
```

Comment:
This is the currently failing test, that really should be prevented by the `jsonschema` bit:

```
        "ongoing": {
            "oneOf": [{
                    "const": false,
                    "required": ["start_date", "end_date"]
                },
                {
                    "const": true,
                    "required": ["start_date"],
                    "not": {
                        "required": ["end_date"]
                    }
                }
            ],
            "type": "boolean"
        },
```

but it isn't and it's confusing!